### DISCUSSION TITLE
Git meminta username dan password untuk beberapa repo

### DISCUSSION THREAD

**Discussion by viechoco** [ditatompel/insights#31] [Q&A] [answered]
Saya ikut tutorial dari @jasmerah1966 untuk setting Git menggunakan SSH di https://insights.ditatompel.com/id/tutorials/how-to-use-git-using-ssh-protocol-for-github/. Tapi beberapa repositori meminta username dan password.

Padahal saat mencoba melakukan test perintah `ssh -T github.com`  pesannya sukses:
```
Hi viechoco! You've successfully authenticated, but GitHub does not provide shell access.
```

Apakah ada langkah terlewat?

Terima kasih

**Comment by ditatompel** [comment, marked as answer]
Kemungkinan itu karena remote repositori di clone dari HTTPS, bukan GIT. Coba cek dari repositori yang selalu minta username pake perintah `git remote -v`.

Contoh output `git remote -v` yang menggunakan HTTPS:
```plain
origin	https://github.com/ditatompel/insights.git (fetch)
origin	https://github.com/ditatompel/insights.git (push)
```

Contoh output  `git remote -v` yang menggunakan SSH:
```plain
origin	git@github.com/ditatompel/insights.git (fetch)
origin	git@github.com/ditatompel/insights.git (push)
```

Remote URL bisa dirubah menggunakan perintah `git remote set-url`. Misal:
```shell
git remote set-url origin git@github.com:ditatompel/insights.git
```

Jd tinggal dirubah bagian `https://github.com` ke  `git@github.com` atau vice versa.

**Reply by viechoco** [reply]
Betul, setelah sy cek ternyata memang karena beberapa repo saya clone dari VScode itu pakai HTTPS.

Sekarang sudah bisa setelah merubah remote urlnya.

Maaci.
